ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1136,21 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    if (!ssl_verify_client_session_version(s)) {
```

Comment:
Oh gosh, you are right. They are error cases only when `ssl_get_new_session` fails. Fixed.